### ISSUE TITLE
Deadlock  using both LoadingPlaceholder and ErrorPlaceholder as gif

### ISSUE THREAD

**Issue by EmilAlipiev** [luberda-molinet/FFImageLoading#717]
I am already using 2.2.10 version
There is indeed similar issue [here](https://github.com/luberda-molinet/FFImageLoading/issues/554)
but maybe this one is fixed, now my situation is another one, not sure. If I have slow connection or no connection, loadingplaceholder will take longer as expected, if I raise imagesource change as shown below, it causes deadlock and app gets frozen in Android.

EDIT: i played around a bit more by changing paramters and I found out that problem is ErrorPlaceholder indeed. if I use;
 - LoadingPlaceHolder : png ErrorPlaceholder: png- No Problem
-  LoadingPlaceHolder : gif ErrorPlaceholder: png- No Problem
- LoadingPlaceHolder : gif  ErrorPlaceholder: gif - Deadlock
-  LoadingPlaceHolder : png ErrorPlaceholder: gif - Deadlock

@daniel-luberda : do you know what could be the problem please? 
```

     <ffimageloading:CachedImage  x:Name="Image1"  	Aspect="AspectFit"    Margin="30,10"  Grid.Row="1" Grid.Column="0"  Grid.ColumnSpan="2"  HorizontalOptions="Center" VerticalOptions="Start" 
                                        LoadingPlaceholder="loading.gif"       RetryCount="0" RetryDelay="0"        CacheDuration = "120" ErrorPlaceholder="error.gif"
                                   Source="{Binding Image1}" AbsoluteLayout.LayoutFlags="All" AbsoluteLayout.LayoutBounds=".5,.5,1,1"
                            />

   private string image1;

        public string Image1
        {
            get { return image1; }
            set { image1 = value; RaisePropertyChanged("Image1"); }
        }
```

**Comment by EmilAlipiev** [comment]
I attach the stacktrace. 
[deadlock_trace.txt](https://github.com/luberda-molinet/FFImageLoading/files/1275518/deadlock_trace.txt)

**Comment by daniel-luberda** [comment]
Could you test the newest prerelease nuget? (2.2.11)

**Comment by EmilAlipiev** [comment]
@daniel-luberda  pre release doesnt crash but it keeps loading and never displays errorplaceholder. I tried using png errorplaceholder with gif loadingplaceholder. png is also not displayed. 2.2.10 displays png errorplaceholder. I have also defined in app.xaml.cs. So expected to time out after 5secs but it keeps loading
```

   var config = new FFImageLoading.Config.Configuration()
            {
                //FadeAnimationDuration = 250,
                //FadeAnimationForCachedImages = true,
                //FadeAnimationEnabled = true,
                HttpHeadersTimeout = 5,
                HttpReadTimeout = 5,
                MaxMemoryCacheSize = 200 * 1024 * 1024                
            };
```

**Comment by daniel-luberda** [comment]
Yes, this is because I forgot to remove this: https://github.com/luberda-molinet/FFImageLoading/commit/13235d8a06e19570f4f7e4744cfdd7932d59e398

I'll update nuget in a few minutes

**Comment by EmilAlipiev** [comment]
I just installed your freshly updated nuget but deadlock appears as in latest stable

**Comment by EmilAlipiev** [comment]
@daniel-luberda please let me know as soon as you have a new update. I will test it again.

**Comment by daniel-luberda** [comment]
It's already pushed. Could you check it?

**Comment by EmilAlipiev** [comment]
@daniel-luberda  I tried latest prerelease unfortunately still freezes the app when using both as gif. I did all delete folder, clean cache, rebuild etc before to make sure I am using correct version

**Comment by daniel-luberda** [comment]
Is it better in the newest stable 2.2.11?

**Comment by daniel-luberda** [comment]
Could you check if it's better in the newest prerelease? There were some fixes / optimisations.

**Comment by daniel-luberda** [comment]
It should be fixed in the newest stable (2.2.12), please reopen if it still occurs.

**Comment by EmilAlipiev** [comment]
Sorry for my late response. Just came back from holiday and tried newest stable 2.2.13. It works fine. Problem looks like solved. Thank you for this fix
